packages/contracts/evm-contracts/contracts/State.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract State {
+    // Price of NFT
+    uint256 public nftPrice;
+
+    bool public initialized;
+
+    // NFT address
+    address public nftAddress;
+
+    // Array of deposited currencies
+    ERC20[] public depositedCurrencies;
+
+    // mapping from token address to amount
```

Comment:
That's not true, it's a mapping from token address to boolean representing if this token has been deposited